app/webpacker/components/ContactsPage/SubForms/Wrt/index.jsx

```diff
@@ -21,15 +21,18 @@ export default function Wrt() {
   );
 
   const QueryForm = useMemo(() => {
-    if (!selectedQueryType) return null;
-    switch (selectedQueryType) {
-      case QUERY_TYPES_MAP.editProfile:
-        return EditProfileQuery;
-      default:
-        return OtherQuery;
-    }
+    if (!selectedQueryType || selectedQueryType === QUERY_TYPES_MAP.editProfile) return null;
+    return OtherQuery;
   }, [selectedQueryType]);
 
+  useEffect(() => {
```

Comment:
I'm not a big fan of `useEffect`. You're trying to react to changes to `selectedQueryType`, so where is this variable being set/written, i.e. where are those changes originally fired? And can you just do what you're doing right here wherever the changes are triggered/fired in the first place?

Reply:
I think I understood what you meant, and I did it in ContactForm.jsx. Please let me know if this is not what you meant.